psd.h

```diff
@@ -201,6 +231,180 @@ namespace psd {
     ChannelID channelID;
     uint64_t length;
   };
+  
+  struct OSType {
+    virtual OSTypeKey type() const = 0;
+    ~OSType(){}
```

Comment:
The destructor must be virtual as this struct contain virtual member functions:
```suggestion
    virtual ~OSType() { }
```